### ISSUE TITLE
[FR] Add compatiblity with transpilers.

### ISSUE THREAD

**Issue by GabrielGorta** [oracle/netsuite-suitecloud-sdk#671]
Let say, I have transpiler, that transform some code in `./src/` directory to other code in `./build/` directory. It would be nice, to be able to trigger upload from `./src/` directory, that would upload transpiled version of code.

Possible solution:

Configuration/setting where user can specify that if they trigger upload in `./src/FileCabined/SuiteScripts/sample.ts` it will upload `./build/FileCabined/SuiteScripts/sample.js`... it could be defined via regex, or syntax like:

```json
{
    "source": ["src/**/*.ts", "src/**/*.js"]
    "build": ["build/**/*.js"]
}
```

or function:

```js
export function pathUpdate(path) {
    return path.replace(/^build/, "src").replace(/.ts$/, ".js");
}
```

**Comment by Rvice** [comment]
Have you tried doing this via the tsconfig file? Create a base tsconfig.json then create your separate build ones

```
  "extends": "./tsconfig.json",
  "compilerOptions": {
	"paths": {
      "N": ["node_modules/@hitc/netsuite-types/N"],
      "N/*": ["node_modules/@hitc/netsuite-types/N/*"],
      "SuiteScripts/*": ["../SuiteScripts/*"]
    },
	"outDir": "./src/FileCabinet/SuiteScripts",
  },
  "exclude": [
	"node_modules",
        ... Other files/folders exclude from this build
   ]
```

then in your package file something like
```
    "build": "npm run build-scripts && npm run build-test",
    "build-scripts": "tsc -p tsconfig.build.json",
    "build-test": "tsc -p tsconfig.tests.json",
```

**Comment by GabrielGorta** [comment]
yes but the issue is that while source files are in `./src/` files, and build files are in `./build/`, I want to be able to execute for example command `suitecloud.uploadfile` without need to opening `./build/` folder. RN, I must navigate to `./build/` as in `./src/` it throws error:

```
The file to upload should be in a supported folder inside the FileCabinet folder. For information about the supported folders, see https://system.netsuite.com/app/help/helpcenter.nl?fid=section_4724963992.html.
```

**Comment by nemchik** [comment]
You may be able to change `defaultProjectFolder` in `suitecloud.config.js` from `src` to `build` (and likely would need a few other adjustments) to accomplish this. I have not tried this.

You can also set up your typescript files in a different subfolder (ex: `./src/FileCabinet/TypeScripts`) and have those transpile into the `./src/FileCabinet/SuiteScripts` folder. This is what I do.

It's up to you if you prefer to commit your build output or not. I prefer to commit the build output so that I have fully tracked files showing exactly what is expected in the NS file cabinet.

There are still quite a few catches with transpiling modern ESM module syntax to legacy AMD module syntax, including typescript itself dropping support for this entirely (requiring the use of an additional tool like swc), so getting usable code for NS from modern typescript can be a struggle, but it's doable (and IMHO, worth doing).

---

Some tips (in case you haven't run into these yet):

DO:

```ts
import * as log from "N/log";
```

DON'T:

```ts
import log from "N/log";
```

DO:

```ts
export { getInputData, map, reduce, summarize }; // for map reduce
```

DON'T:

```ts
export default { getInputData, map, reduce, summarize };
```

The rest should fall into place if you're using [@hitc/netsuite-types](https://www.npmjs.com/package/@hitc/netsuite-types)
